services/dedup/mirror.go

```diff
@@ -44,85 +44,112 @@ func newMirror(d, m Dedup, conf *config.Config, s stats.Stats, log logger.Logger
 	group := &errgroup.Group{}
 	group.SetLimit(groupLimit)
 
-	dedupMirror := &mirror{
-		Dedup:         d,
-		mirror:        m,
+	mirrorDB := &MirrorDB{
+		primary:       primary,
+		mirror:        mirror,
+		mode:          mode,
 		group:         group,
-		groupLimit:    groupLimit, // storing limit for enriching error
+		groupLimit:    groupLimit,
 		errs:          make(chan error, 1),
 		stopPrintErrs: make(chan struct{}),
 		logger:        log,
 	}
-	dedupMirror.metrics.allowedErrorsCount = s.NewTaggedStat("dedup_mirroring_err_count", stats.CountType, stats.Tags{
-		"mode": "keydb",
-		"type": "allowed",
+
+	// Set up metrics
+	mirrorModeStr := "keydb"
+	if mode == mirrorBadgerMode {
+		mirrorModeStr = "badgerdb"
+	}
```

Comment:
We don't need this here since it shouldn't be aware of the various modes. Either use the original `Mode` string, or introduce a `MirrorLabel` method in `Mode` that knows how to return the label.
```suggestion
```